doc/source/user_guide/visualization.rst

```diff
@@ -821,7 +821,7 @@ You can create a scatter plot matrix using the
    df = pd.DataFrame(np.random.randn(1000, 4), columns=['a', 'b', 'c', 'd'])
 
    @savefig scatter_matrix_kde.png
-   scatter_matrix(df, alpha=0.2, figsize=(6, 6), diagonal='kde')
+   scatter_matrix(df, alpha=0.2, figsize=(6, 6), diagonal='kde');
```

Comment:
Does using plt.show() have the same effect? May look a bit more natural.

Reply:
I am not sure how to do this using plt.show()

Reply:
I think this is fine. IPython will show any output of each line. So adding a `plt.show()` will not prevent printing the output of this line.

Reply:
This seems like a strange one-off so would prefer what @dsaxton suggests if it works



> IPython will show any output of each line.

Is that a general comment or specific to just doc rendering? Running this in ipython:

```python
a = [1, 2, 3]
a
print("done")
```

Only prints "done"

Reply:
> Is that a general comment or specific to just doc rendering?

The IPython directive runs line by line (so it's not the same copy pasting it in an ipython console, but rather as copy pasting line by line)

> This seems like a strange one-off 

We already do this in several places in this `visualization.rst`, so it's not a one-off

Reply:
Ah OK thanks for the explanation. lgtm to then

Reply:
there are about 40 semicolons in visualization.rst but there are also many plots which don't have semicolons and show the debug messages.
Should we put a semicolon after every plot in visualization.rst?